### PR TITLE
refactor view for creating new test run, use class based view

### PR DESCRIPTION
**Pull request by RMadjev**

### CODE REVIEW

File: tcms/testruns/views.py [hunk 1]
```diff
@@ -38,94 +38,96 @@
...
-    test_plan = TestPlan.objects.get(plan_id=plan_id)
-
-    # note: ordered by case_id for test_show_create_new_run_page()
-    tcs_values = test_cases.select_related('author',
-                                           'case_status',
-                                           'category',
-                                           'priority').order_by('case_id')
+@method_decorator(permission_required('testruns.add_testrun'), name='dispatch')
+class CreateTestRunView(View):
+    """Create new test run"""
```
Comment (atodorov) [inline tcms/testruns/views.py]: Missing `http_method_names` attribute

File: tcms/testruns/views.py [hunk 2]
```diff
@@ -38,94 +38,96 @@
...
+                    manager=form.cleaned_data['manager'],
+                    default_tester=default_tester,
+                )
 
-            loop = 1
-            for case in form.cleaned_data['case']:
                 try:
-                    tcp = TestCasePlan.objects.get(plan=test_plan, case=case)
-                    sortkey = tcp.sortkey
+                    assignee_tester = User.objects.get(username=default_tester)
```
Comment (atodorov) [inline tcms/testruns/views.py]: Remove this entire block around `assignee_tester` and replace this variable with `User.objects.filter(username=default_tester).first()` below (this will reduce number of local variables and complexity)

File: tcms/testruns/views.py [hunk 3]
```diff
@@ -38,94 +38,96 @@
...
+        if not request.POST.get('from_plan'):
+            messages.add_message(request,
+                                 messages.ERROR,
+                                 _('Creating a TestRun requires a TestPlan, select one'))
+            return HttpResponseRedirect(reverse('plans-search'))
+
+        plan_id = request.POST.get('from_plan')
+        # case is required by a test run
+        # NOTE: currently this is handled in JavaScript but in the TestRun creation
+        # form cases can be deleted
```
Comment (atodorov) [inline tcms/testruns/views.py]: remove this comment b/c it is not correct anymore.

File: tcms/testruns/views.py [hunk 4]
```diff
@@ -38,94 +38,96 @@
...
+                'notes': '',
+            })
+            form.populate(product_id=test_plan.product_id)
+
+        context_data = {
+            'test_plan': test_plan,
+            'test_cases': tcs_values,
+            'form': form,
+        }
+        return render(request, 'testruns/mutable.html', context_data)
```
Comment (atodorov) [inline tcms/testruns/views.py]: Move the template name into a `template_name` class attribute. This will make it easier to transition to TemplateView or FormView later if we decide to.

File: tcms/testruns/views.py [hunk 5]
```diff
@@ -38,94 +38,96 @@
...
-    # Ready to write cases to test plan
-    test_cases = get_selected_testcases(request)
-    test_plan = TestPlan.objects.get(plan_id=plan_id)
-
-    # note: ordered by case_id for test_show_create_new_run_page()
-    tcs_values = test_cases.select_related('author',
-                                           'case_status',
-                                           'category',
-                                           'priority').order_by('case_id')
+@method_decorator(permission_required('testruns.add_testrun'), name='dispatch')
```
Comment (atodorov) [inline tcms/testruns/views.py]: Wrt permission tests: from what I can see we have tests which include the permission label (these need to be more granual - include only that permission label and nothing else). We're missing tests for negative scenario where the user doesn't have permission assigned.

**Comment by RMadjev** [comment]
Please ignore the last commit.
